nextcord/http.py

```diff
@@ -124,25 +140,329 @@ def bucket(self) -> str:
+            )
+
+        # Updates the datetime of the reset.
+        x_reset = response.headers.get("X-RateLimit-Reset")
+        if x_reset is not None:
+            x_reset = datetime.fromtimestamp(float(x_reset) + self._time_offset, tz=timezone.utc)
+
+            if self._use_reset_timestamp and self.reset is not None:
+                new_reset_diff = (x_reset - self.reset).total_seconds()
+                if self._tracked_reset_time < new_reset_diff:
+                    # Any time the tracked reset time is increased, the reset timestamp should also be increased
+                    #  thus we shouldn't have to (re)start the reset task here.
+                    self._tracked_reset_time = new_reset_diff
+
+            if self.reset is None or self.reset < x_reset:
+                self.reset = x_reset
+                if self._use_reset_timestamp:
+                    _log.debug(
+                        "Bucket %s: Reset timestamp increased, starting/resetting reset task.",
+                        self.bucket,
+                    )
+                    self.start_reset_task()
+
+        # Updates the reset-after count, being pessimistic.
+        x_reset_after = response.headers.get("X-RateLimit-Reset-After")
+        if x_reset_after is not None:
+            x_reset_after = float(x_reset_after) + self._time_offset
+            self.reset_after = x_reset_after
+            # Once we figure out what the true reset delay is, for example 5 seconds, we want to keep it at that.
+            if not self._use_reset_timestamp and self._tracked_reset_time < x_reset_after:
+                self._tracked_reset_time = x_reset_after
+                _log.debug(
+                    "Bucket %s: Reset after time increased to %s seconds, adapting reset time.",
+                    self.bucket,
+                    self._tracked_reset_time,
+                )
+                self.start_reset_task()
+
+        if not self.resetting:
+            self.start_reset_task()
+
+        # If for whatever reason we have requests remaining but the reset event isn't set, set it.
+        if self.remaining > 0 and not self._on_reset_event.is_set():
+            _log.debug(
+                "Bucket %s: Updated with remaining %s, setting reset event.",
+                self.bucket,
+                self.remaining,
+            )
+            self._on_reset_event.set()
+
+        # If this is our first update, indicate that all future updates aren't the first.
+        if self._first_update:
+            self._first_update = False
+
+        _log.debug(
+            "Bucket %s: Updated with limit %s, remaining %s, reset %s, and reset_after %s seconds.",
+            self.bucket,
+            self.limit,
+            self.remaining,
+            self.reset,
+            self.reset_after,
+        )
+
+    def start_reset_task(self) -> None:
+        """Starts the reset task, non-blocking."""
```

Comment:
This should be private.
Also no need to comment it is non blocking? Document just the blocking parts imo